src/celeritas/optical/OpticalGenData.hh

```diff
@@ -90,11 +90,12 @@ struct OpticalPreStepData
     units::LightSpeed speed;
     Real3 pos{};
     real_type time{};
+    OpticalMaterialId opt_mat;
 
     //! Check whether the data are assigned
     explicit CELER_FUNCTION operator bool() const
     {
-        return speed > zero_quantity();
+        return speed > zero_quantity() && opt_mat;
```

Comment:
It might be slightly more efficient to flip the order of these? Or even just test for `opt_mat`. Since we're going to be evaluating this on every thread once or twice per step...

Reply:
I'll flip the order; we'll need to check the speed to make sure the particle didn't start the step with zero energy (I removed the `sim.step_length() == 0` check and replaced it with this).